internal/proxyutil/psdb.go

```diff
@@ -0,0 +1,27 @@
+package proxyutil
+
+import (
+	"log/slog"
+
+	"github.com/planetscale/psdb/auth"
+	"github.com/planetscale/psdbproxy"
+	"go.uber.org/zap"
+	"go.uber.org/zap/exp/zapslog"
+)
+
+type Config struct {
+	Logger       *zap.Logger
+	LocalAddr    string
+	UpstreamAddr string
+	Username     string
+	Password     string
+}
+
+func New(cfg Config) *psdbproxy.Server {
+	return &psdbproxy.Server{
+		Addr:          cfg.LocalAddr,
+		Logger:        slog.New(zapslog.NewHandler(cfg.Logger.Core(), nil)),
+		UpstreamAddr:  cfg.UpstreamAddr,
+		Authorization: auth.NewBasicAuth(cfg.Username, cfg.Password),
+	}
+}
```

Comment:
I don't think we need this extra package for constructing the server type. I would include this only if we reuse it in multiple places. Otherwise it'll just create an additional abstraction with no immediate benefits.

Reply:
For context, #760 #761 and #762 are the other implementations of this.